src/client/metric.rs

```diff
@@ -286,11 +292,169 @@ impl<T: MetricType + Clone> Metric<T> {
         self.mmap_view = mmap_view;
     }
 
-    pub (super) fn write_val(&self, cursor: &mut Cursor<&mut [u8]>) -> io::Result<()> {
-        self.val.write_to_writer(cursor)
+    pub (super) fn write_val<W: Write>(&self, writer: &mut W) -> io::Result<()> {
+        self.val.write_to_writer(writer)
     }
 }
 
+#[derive(Clone)]
+/// An instance domain is a set of instances
+pub struct Indom {
+    pub (super) instances: HashSet<String>,
+    pub (super) id: u32,
+    shorthelp: String,
```

Comment:
Advantage would be the user not having to keep the passed `str` in the same or larger scope that `Indom` lives in. Disadvantage would be the extra allocation.